_api_app/app/Sites/SitesMenuRenderService.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace App\Sites;
+
+use App\Shared\Helpers;
+use App\Shared\Storage;
+use App\Sites\SitesDataService;
+use App\Sites\Settings\SiteSettingsDataService;
+
+class sitesMenuRenderService
+{
+    private $currentSite;
+    private $isEditMode;
+    private $siteTemplateSettings;
+
+    /**
+     * Construct SitesRenderService instance
+     *
+     * @param string $currentSite
+     * @param bool $isEditMode
+     * @param array $siteTemplateSettings
+     */
+    public function __construct(
+        $currentSite,
+        $isEditMode,
+        array $siteSettings,
+        array $siteTemplateSettings
+    ) {
+        $this->currentSite = $currentSite;
+        $this->isEditMode = $isEditMode;
+        $this->siteSettings = $siteSettings;
+        $this->siteTemplateSettings = $siteTemplateSettings;
+    }
+
+    public function getStyles($params)
+    {
+        $pos = !empty($params) ? explode(',', $params) :
+            [
+                rand(0, 960),
+                rand(0, 600)
+            ];
+        return 'left:' . $pos[0] . 'px;top:' . $pos[1] . 'px;';
+    }
+
+    public function getViewData()
+    {
+        $sitesDataService = new SitesDataService();
```

Comment:
it would be better to inject the `SiteDataService` instance into the constructor.